### PR TITLE
chore(deps): bump phoenix

### PR DESCRIPTION
**Pull request by JucanAndreiDaniel**
Update Phoenix to the current main branch

### CODE REVIEW

**Comment by lmichaelis** [comment]
We need to either replace the `PxVdf` implementation with the new `phoenix::Vfs` class or allow deprecated symbols for now. I'm removing the old API from phoenix in v2 which I intend to release later this year.

**Comment by JaXt0r** [comment]
@lmichaelis 

VDF --> VFS. Mostly only renaming. Changes to check closely are inside:
* Vfs.h
* Vfs.cc

File: include/phoenix/cffi/Vfs.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	#include <phoenix/Vfs.hh>
+typedef phoenix::Vfs PxVfs;
+typedef phoenix::VfsNode PxVfsNode;
+#else
+typedef struct PxInternal_Vfs PxVfs;
+typedef struct PxInternal_VfsNode PxVfsNode;
+#endif
+
+PXC_API PxVfs* pxVfsNew();
+PXC_API void pxVfsMountFile(PxVfs* vfs, char const* path);
```
Comment (lmichaelis) [inline include/phoenix/cffi/Vfs.h]: This should be called `pxVfsMountDisk`

File: src/Vfs.cc [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// Copyright © 2023. GothicKit Contributors
+// SPDX-License-Identifier: MIT
+#include "Prelude.h"
+
+#include <phoenix/Vfs.hh>
+#include <phoenix/cffi/Vfs.h>
+
+PxVfs* pxVfsNew() {
+	return RC(PxVfs, new phoenix::Vfs());
```
Comment (lmichaelis) [inline src/Vfs.cc]: This `RC(...)` wrapper is not required since `PxVfs` is already an alias for `phoenix::Vfs`.

**Review by lmichaelis — CHANGES_REQUESTED** [review]
In addition, please update the phoenix submodule again because [phoenix/bcb47c1](https://github.com/GothicKit/phoenix/commit/bcb47c1b545962efdc205db9494d1d734d1bed11) fixes an issue I introduced in a recent commit which only happens in release builds.

**Comment by JaXt0r** [comment]
Changes implemented. Please recheck.

**Pull request merged**